source/postprocess/visualization/volume_of_fluid_values.cc

```diff
@@ -0,0 +1,201 @@
+      }
+
+
+      template <int dim>
+      void
+      VolumeOfFluidValues<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double>> &computed_quantities) const
+      {
+        const unsigned int n_quadrature_points = input_data.solution_values.size();
+        Assert (computed_quantities.size() == n_quadrature_points, ExcInternalError ());
+        Assert (input_data.solution_values[0].size() == this->introspection().n_components, ExcInternalError ());
+
+        unsigned int out_per_field=1;
+        if (include_contour)
+          out_per_field += 1;
+        if (include_normal)
+          out_per_field += dim;
+
+        for (unsigned int f=0; f<this->get_volume_of_fluid_handler().get_n_fields(); ++f)
+          {
+            VolumeOfFluidField<dim> field = this->get_volume_of_fluid_handler().field_struct_for_field_index(f);
+
+            const FEVariable<dim> &volume_of_fluid_var = field.volume_fraction;
+            const unsigned int volume_of_fluid_ind = volume_of_fluid_var.first_component_index;
+            const FEVariable<dim> &volume_of_fluidLS_var = field.level_set;
+            const unsigned int volume_of_fluidLS_ind = volume_of_fluidLS_var.first_component_index;
+
+            for (unsigned int q=0; q<n_quadrature_points; ++q)
+              {
+                unsigned int out_ind = f*out_per_field;
+                computed_quantities[q][out_ind] = input_data.solution_values[q][volume_of_fluid_ind];
+                ++out_ind;
+                if (include_contour)
+                  {
+                    computed_quantities[q][out_ind] = input_data.solution_values[q][volume_of_fluidLS_ind];
+                    ++out_ind;
+                  }
+
+                if (include_normal)
+                  {
+                    Tensor<1, dim, double> normal = -input_data.solution_gradients[q][volume_of_fluidLS_ind];
+                    for (unsigned int i = 0; i<dim; ++i)
+                      {
+                        computed_quantities[q][out_ind] = normal[i];
+                        ++out_ind;
+                      }
+                  }
+              }
+          }
+      }
+
+
+      template <int dim>
+      void
+      VolumeOfFluidValues<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Visualization");
+          {
+            prm.enter_subsection("Volume of Fluid");
+            {
+              prm.declare_entry("Include internal reconstruction contour", "false",
+                                Patterns::Bool (),
+                                "Include the internal level set data use to save reconstructed interfaces");
+
+              prm.declare_entry("Include normals", "false",
+                                Patterns::Bool (),
+                                "Include internal normal data in output (DEBUG)");
```

Comment:
Why is this DEBUG? I think this might be useful. Please remove the DEBUG marker.

Reply:
The main issue is that currently this does not adjust the output to match the real cell, but remains on the unit cell.

Reply:
To clarify: if we were working on a mesh where the unit cell directions did not match the actual cell (eg due to rotations of the mesh), the normal vector would be in terms of the unit cell x and y, not the "physical space"

Reply:
Ok, then write that into the documentation, but remove the debug. At the moment this can not happen anyway, as you only allow the box geometry. When this is extended to spherical models one just needs to find this place. Also add a comment like the following above the parameter:
`// TODO: Fix this for curved geometries`